client/lib/use-in-view/index.ts

```diff
@@ -0,0 +1,46 @@
+import { RefObject, useEffect, useRef } from 'react';
+
+export function useInView< T extends Element >( oneTimeCallback: () => void ): RefObject< T > {
+	const viewedRef = useRef< boolean >( false );
+	const elementRef = useRef< T >( null );
+	const observerRef = useRef< IntersectionObserver >();
+
+	useEffect( () => {
+		// We can't do anything without a valid reference to an element on the page
+		if ( ! elementRef.current ) {
+			return;
+		}
+
+		// If the observer is already defined, no need to continue
+		if ( observerRef.current ) {
+			return;
+		}
+
+		const handler = ( entries: IntersectionObserverEntry[] ) => {
+			// Only fire once per page load
+			if ( viewedRef.current ) {
```

Comment:
What if someone wants this to fire every time the element is in view?

Reply:
It would require an update but just a small one. Do we have that use case currently?

Reply:
Not sure, haven't looked through the codebase.

Personally, I realize I'm strongly against reinventing the wheel in this case, so I'm going to refrain from reviewing this pull request further.